trie/patriciaMerkleTrie_test.go

```diff
@@ -997,84 +997,141 @@ func TestPatriciaMerkleTree_GetValueReturnsTrieDepth(t *testing.T) {
-			case 8:
-				trieIteratorChannels := &common.TrieIteratorChannels{
-					LeavesChan: make(chan core.KeyValueHolder, 1000),
-					ErrChan:    errChan.NewErrChanWrapper(),
+		for i := 0; i < numOperations; i++ {
+			go func(idx int) {
+				time.Sleep(time.Millisecond * 10)
+
+				operation := idx % numFunctions
+				switch operation {
+				case 0:
+					_, _, err := tr.Get([]byte("dog"))
+					assert.Nil(t, err)
+				case 1:
+					tr.Update([]byte("doe"), []byte("alt"))
+				case 2:
+					tr.Delete([]byte("alt"))
+				case 3:
+					_, err := tr.RootHash()
+					assert.Nil(t, err)
+				case 4:
+					err := tr.Commit(hashesCollector.NewDisabledHashesCollector())
+					assert.Nil(t, err)
+				case 5:
+					epoch := core.OptionalUint32{
+						Value:    3,
+						HasValue: true,
+					}
+					rootHashHolder := holders.NewRootHashHolder(initialRootHash, epoch)
+					_, err := tr.Recreate(rootHashHolder, "")
+					assert.Nil(t, err)
+				case 6:
+					_, err := tr.GetSerializedNode(initialRootHash)
+					assert.Nil(t, err)
+				case 7:
+					size1KB := uint64(1024 * 1024)
+					_, _, err := tr.GetSerializedNodes(initialRootHash, size1KB)
+					assert.Nil(t, err)
+				case 8:
+					trieIteratorChannels := &common.TrieIteratorChannels{
+						LeavesChan: make(chan core.KeyValueHolder, 1000),
+						ErrChan:    errChan.NewErrChanWrapper(),
+					}
+
+					err := tr.GetAllLeavesOnChannel(
+						trieIteratorChannels,
+						context.Background(),
+						initialRootHash,
+						keyBuilder.NewKeyBuilder(),
+						parsers.NewMainTrieLeafParser(),
+					)
+					assert.Nil(t, err)
+				case 9:
+					_, _, _ = tr.GetProof(initialRootHash, initialRootHash) // this might error due to concurrent operations that change the roothash
+				case 10:
+					// extremely hard to compute an existing hash due to concurrent changes.
+					_, _ = tr.VerifyProof([]byte("dog"), []byte("puppy"), [][]byte{[]byte("proof1")}) // this might error due to concurrent operations that change the roothash
+				case 11:
+					sm := tr.GetStorageManager()
+					assert.NotNil(t, sm)
+				case 12:
+					trieStatsHandler := tr.(common.TrieStats)
+					_, err := trieStatsHandler.GetTrieStats("address", initialRootHash)
+					assert.Nil(t, err)
+				default:
+					assert.Fail(t, fmt.Sprintf("invalid numFunctions value %d, operation: %d", numFunctions, operation))
 				}
 
-				err := tr.GetAllLeavesOnChannel(
-					trieIteratorChannels,
-					context.Background(),
-					initialRootHash,
-					keyBuilder.NewKeyBuilder(),
-					parsers.NewMainTrieLeafParser(),
-				)
-				assert.Nil(t, err)
-			case 9:
-				_, _, _ = tr.GetProof(initialRootHash, initialRootHash) // this might error due to concurrent operations that change the roothash
-			case 10:
-				// extremely hard to compute an existing hash due to concurrent changes.
-				_, _ = tr.VerifyProof([]byte("dog"), []byte("puppy"), [][]byte{[]byte("proof1")}) // this might error due to concurrent operations that change the roothash
```

Comment:
are these comments still valid now with the latest changes?

Reply:
No, removed the comments and added some asserts